sensor_test.py

```diff
@@ -0,0 +1,214 @@
+    """
+    1. Set sensor name to "new_name".
+    2. Get sensor_info.
+    3. Validate that current sensor name matches the name set in Step 1.
+    """
+    print("Set sensor name to 'new_name'")
+    set_sensor_name("new_name")
+
+    print("Get sensor info")
+    sensor_info = get_sensor_info()
+
+    print(
+        f"Validate that current sensor name matches the name set in Step 1, current name is: {sensor_info['name']}"
+    )
+    assert sensor_info["name"] == "new_name", "Sensor name was not set correctly"
+
+
+def test_set_sensor_reading_interval(
+    get_sensor_info, set_sensor_reading_interval, get_sensor_reading
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+    print("1. Set sensor reading interval to 1")
+    interval = 1
+    set_sensor_reading_interval(interval)
+
+    print("2. Get sensor info")
+    sensor_info = get_sensor_info()
+
+    print("3. Validate that sensor reading interval is set to interval from Step 1")
+    assert (
+        sensor_info["reading_interval"] == interval
+    ), "Sensor reading interval was not set correctly"
+
+    print("4. Get sensor reading")
+    sensor_reading = get_sensor_reading()
+
+    print("5. Wait for interval specified in Step 1")
+
+    print("6. Get sensor reading")
+    sensor_reading_after_interval = wait(
+        func=get_sensor_reading(),
+        condition=lambda x: isinstance(x, float),
+        tries=15,
+        timeout=interval,
+    )
+
+    print(
+        f"Validate that reading from Step 4 doesn't equal reading from Step 6, current readings are: {sensor_reading} and {sensor_reading_after_interval}"
+    )
+    assert (
+        sensor_reading != sensor_reading_after_interval
+    ), "Sensor reading interval is not working correctly"
+
+
+# Максимальна версія прошивки сенсора -- 15,
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+
+    max_firmware_version = 15
+    firmware_version = "firmware_version"
+
+    print("Get original sensor firmware version")
+    sensor_info = get_sensor_info()
+    initial_firmware_version = sensor_info[firmware_version]
```

Comment:
This is a well-named variable. However, in the context it's used further down its name can be confusing.

`initial_firmware_version` assumes that this is a fixed value, a reference if you will.

So, if it gets updated in the loop below, the name loses its meaning.

To solve this, you could go two ways:
1. Rename this variable to `current_firmware_version`.
2. Add another variable below it like so: `current_firmware_version = initial_firmware_version` and in your code update `current_firmware_version`.